src/Proto3/Suite/JSONPB/Instances.hs

```diff
@@ -0,0 +1,183 @@
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE OverloadedLists     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- | Common instances for jsonpb codec implementations
+
+module Proto3.Suite.JSONPB.Instances where
+
+import qualified Data.Aeson                as A (ToJSON (..), Value (..))
+import qualified Data.Aeson.Encoding       as E
+import qualified Data.Aeson.Types          as A (typeMismatch)
+import qualified Data.ByteString           as BS
+import qualified Data.ByteString.Base64    as B64
+import qualified Data.Text.Encoding        as T
+import qualified Data.Text.Lazy            as TL
+import qualified Data.Vector               as V
+import           GHC.Int                   (Int32, Int64)
+import           GHC.Word                  (Word32, Word64)
+import           Proto3.Suite.Class        (HasDefault (..), Named (..))
+import           Proto3.Suite.JSONPB.Class
+import           Proto3.Suite.Types        (Enumerated (..), Fixed (..))
+
+-- * Instances for scalar types
+
+--------------------------------------------------------------------------------
+-- Boolean scalar type
+
+instance ToJSONPB Bool
+instance FromJSONPB Bool
+
+--------------------------------------------------------------------------------
+-- Integer scalar types
+--
+--   * 32 bit integer values render to JSON decimal numbers; either numbers or
+--     strings are accepted.
+--
+--   * 64 bit integer values render to JSON decimal strings; either numbers
+--     or strings are accepted.
+--
+
+-- int32 / sint32
+instance ToJSONPB Int32
+instance FromJSONPB Int32 where
+  parseJSONPB = parseNumOrDecimalString "int32 / sint32"
+
+-- uint32
+instance ToJSONPB Word32
+instance FromJSONPB Word32 where
+  parseJSONPB = parseNumOrDecimalString "uint32"
+
+-- int64 / sint64
+instance ToJSONPB Int64 where
+  toEncodingPB _ = E.string . show
+instance FromJSONPB Int64 where
+  parseJSONPB = parseNumOrDecimalString "int64 / sint64"
+
+-- unit64
+instance ToJSONPB Word64 where
+  toEncodingPB _ = E.string . show
+instance FromJSONPB Word64 where
+  parseJSONPB = parseNumOrDecimalString "int64 / sint64"
+
+-- fixed32
+instance ToJSONPB (Fixed Word32) where
+  toEncodingPB opts = toEncodingPB opts . fixed
+instance FromJSONPB (Fixed Word32) where
+  parseJSONPB = fmap Fixed . parseJSONPB
+
+-- fixed64
+instance ToJSONPB (Fixed Word64) where
+  toEncodingPB opts = toEncodingPB opts . fixed
+instance FromJSONPB (Fixed Word64) where
+  parseJSONPB = fmap Fixed . parseJSONPB
+
+-- sfixed32
+instance ToJSONPB (Fixed Int32) where
+  toEncodingPB opts = toEncodingPB opts . fixed
+instance FromJSONPB (Fixed Int32) where
+  parseJSONPB = fmap Fixed . parseJSONPB
+
+-- sfixed64
+instance ToJSONPB (Fixed Int64) where
+  toEncodingPB opts = toEncodingPB opts . fixed
```

Comment:
For all of these instances that just lift `toEncodingPB` over a newtype wrapper, you can use:

```haskell
toEncodingPB = coerce (toEncodingPB @Int64)
```